### ISSUE TITLE
'WrappedSocket' object has no attribute 'close'

### ISSUE THREAD

**Issue by DamienCassou** [matlink/gplaycli#15]
```
$ gplaycli --verbose -d com.sncf.fusion -f ~/Documents/projects/playstore/
1/1 com.sncf.fusion
Error while downloading com.sncf.fusion : 'WrappedSocket' object has no attribute 'close'
```

**Comment by matlink** [comment]
I can't reproduce that error. Could you retry ?
My `./gplaycli.py --verbose -d com.sncf.fusion -p` worked well.

**Comment by DamienCassou** [comment]
If I change the code to throw the download exception, I get:

```
Traceback (most recent call last):
  File "/nix/store/dvsqrgi6lkk5yw4npsln0vnb86jz5fms-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 345, in <module>
    main()
  File "/nix/store/dvsqrgi6lkk5yw4npsln0vnb86jz5fms-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 342, in main
    cli.download_packages(args.packages_to_download)
  File "/nix/store/dvsqrgi6lkk5yw4npsln0vnb86jz5fms-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 262, in download_packages
    self.download_selection(self.playstore_api, [(pkg,None) for pkg in list_of_packages_to_download], self.after_download)
  File "/nix/store/dvsqrgi6lkk5yw4npsln0vnb86jz5fms-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 180, in download_selection
    raise(exc)
AttributeError: 'WrappedSocket' object has no attribute 'close'
```

**Comment by DamienCassou** [comment]
If I remove the exception block, I get:

```
Traceback (most recent call last):
  File "/nix/store/7bzagh2w01fih5vallgnshj87i3rqpgz-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 341, in <module>
    main()
  File "/nix/store/7bzagh2w01fih5vallgnshj87i3rqpgz-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 338, in main
    cli.download_packages(args.packages_to_download)
  File "/nix/store/7bzagh2w01fih5vallgnshj87i3rqpgz-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 258, in download_packages
    self.download_selection(self.playstore_api, [(pkg,None) for pkg in list_of_packages_to_download], self.after_download)
  File "/nix/store/7bzagh2w01fih5vallgnshj87i3rqpgz-python2.7-gplaycli-git/bin/.gplaycli-wrapped", line 174, in download_selection
    data = playstore_api.download(packagename, vc,progress_bar=self.progress_bar)
  File "/nix/store/7bzagh2w01fih5vallgnshj87i3rqpgz-python2.7-gplaycli-git/lib/python2.7/site-packages/ext_libs/googleplay_api/googleplay.py", line 285, in download
    response = requests.get(url, headers=headers, cookies=cookies, verify=ssl_verify,stream=True)
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/api.py", line 55, in get
    return request('get', url, **kwargs)
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/api.py", line 44, in request
    return session.request(method=method, url=url, **kwargs)
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/sessions.py", line 335, in request
    resp = self.send(prep, **send_kwargs)
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/sessions.py", line 454, in send
    history = [resp for resp in gen] if allow_redirects else []
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/sessions.py", line 144, in resolve_redirects
    allow_redirects=False,
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/sessions.py", line 438, in send
    r = adapter.send(request, **kwargs)
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/adapters.py", line 292, in send
    timeout=timeout
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 428, in urlopen
    body=body, headers=headers)
  File "/nix/store/bdv10gid5sn2m8vihjggngcwyh8dwaxn-python2.7-requests-1.2.3/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 288, in _make_request
    httplib_response = conn.getresponse(buffering=True)
  File "/nix/store/v9pq6f0s1r5fdybqc7pbv7mkb33lx9yy-python-2.7.10/lib/python2.7/httplib.py", line 1138, in getresponse
    self.close()
  File "/nix/store/v9pq6f0s1r5fdybqc7pbv7mkb33lx9yy-python-2.7.10/lib/python2.7/httplib.py", line 844, in close
    sock.close()   # close it manually... there may be other refs
AttributeError: 'WrappedSocket' object has no attribute 'close'
```

**Comment by DamienCassou** [comment]
@matlink Unfortunately, the links don't help: https://github.com/nviennot/android-checkin/issues/3 https://github.com/nviennot/android-checkin/issues/4 https://github.com/Akdeniz/google-play-crawler/issues/80

Looks like I'm very unlucky

**Comment by DamienCassou** [comment]
This issue is similar to https://codingteam.net/project/googleplaydownloader/bugs/show/3052

**Comment by DamienCassou** [comment]
I managed to generate a new androidid but now face https://github.com/matlink/gplaycli/issues/16

**Comment by matlink** [comment]
Seems like your application is not compatible with you android device 
-- 
Matlink - sysadmin Matlink.fr

**Comment by DamienCassou** [comment]
@matlink I generated the androidid using googleplaydownloader. I thought these IDs represent newest devices.

**Comment by matlink** [comment]
No, It generates a fixed device AndroidID. I don't remember which one. 
Whatever, the default credentials should work, I can download capitain train by myself (I use it too, it's a great app BTW).

**Comment by DamienCassou** [comment]
I can still confirm the issue when using default credentials:

```
$ gplaycli --verbose -d com.sncf.fusion -p -c /etc/gplaycli/credentials.conf 
1/1 com.sncf.fusion
	Error while downloading com.sncf.fusion : 'WrappedSocket' object has no attribute 'close'
```

**Comment by raphael0202** [comment]
I just cloned the repository, and I don't have this issue (I could download the app). Which AndroidID do you use?
Mine is the default:
```[Credentials]
gmail_password=jesuischarlie1
android_ID=3d716411bf8bc802
gmail_address=fdroidserver@gmail.com
language=fr_FR
```

**Comment by DamienCassou** [comment]
I'm using default credentials

**Comment by DamienCassou** [comment]
```
$ gplaycli --config ~/Documents/projects/gplaycli/credentials.conf --folder ~/Documents/projects/playstore/ --download com.capitainetrain.android --progress
Error while downloading com.capitainetrain.android : 'WrappedSocket' object has no attribute 'close'

$ cat ~/Documents/projects/gplaycli/credentials.conf 
[Credentials]
gmail_password=jesuischarlie1
android_ID=3d716411bf8bc802
gmail_address=fdroidserver@gmail.com
language=fr_FR
```

**Comment by matlink** [comment]
Solved by #20 .
